cmake/modules/FindP4EST.cmake

```diff
@@ -123,7 +129,7 @@ IF(EXISTS ${P4EST_INCLUDE_DIR}/p4est_config.h)
   # Is p4est built against zlib?
   #
   FILE(STRINGS "${P4EST_INCLUDE_DIR}/p4est_config.h" P4EST_ZLIB_STRING
-    REGEX "#define.*P4EST_HAVE_ZLIB 1")
+    REGEX "^#define.*P4EST_HAVE_ZLIB")
```

Comment:
Not important, but the C preprocessor syntax doesn't actually require the `#` to appear at the top of the line (it just has to be the first non-space-non-tab of the line), and it can be followed by zero or more spaces. So technically, the regex isn't quite correct. But then if it works, that's good enough :-)